crates/muvm/src/launch.rs

```diff
@@ -59,53 +60,69 @@ pub fn launch_or_lock(
     match lock_file {
         Some(lock_file) => Ok(LaunchResult::LockAcquired {
+            cookie,
             lock_file,
             command,
             command_args,
             env,
         }),
         None => {
-            if let Some(port) = running_server_port {
-                let env = prepare_env_vars(env)?;
-                let mut tries = 0;
-                loop {
-                    match request_launch(port, command.clone(), command_args.clone(), env.clone()) {
-                        Err(err) => match err.downcast_ref::<LaunchError>() {
-                            Some(&LaunchError::Connection(_)) => {
-                                if tries == 3 {
-                                    return Err(anyhow!(
-                                        "could not request launch to server: {err}"
-                                    ));
-                                } else {
-                                    tries += 1;
-                                }
-                            },
-                            _ => {
+            let env = prepare_env_vars(env)?;
+            let mut tries = 0;
+            loop {
+                match request_launch(
+                    server_port,
+                    cookie,
+                    command.clone(),
+                    command_args.clone(),
+                    env.clone(),
+                ) {
+                    Err(err) => match err.downcast_ref::<LaunchError>() {
+                        Some(&LaunchError::Connection(_)) => {
+                            if tries == 3 {
                                 return Err(anyhow!("could not request launch to server: {err}"));
-                            },
+                            } else {
+                                tries += 1;
+                            }
                         },
-                        Ok(_) => return Ok(LaunchResult::LaunchRequested),
-                    }
+                        _ => {
+                            return Err(anyhow!("could not request launch to server: {err}"));
+                        },
+                    },
+                    Ok(_) => return Ok(LaunchResult::LaunchRequested),
                 }
-            } else {
-                Err(anyhow!(
-                    "muvm is already running but couldn't find its server port, bailing out"
-                ))
             }
         },
     }
 }
 
-fn lock_file(server_port: u32) -> Result<(Option<File>, Option<u32>)> {
+fn read_cookie() -> Result<Uuid> {
+    let run_path = env::var("XDG_RUNTIME_DIR")
+        .context("Failed to read XDG_RUNTIME_DIR environment variable")?;
+    let lock_path = Path::new(&run_path).join("muvm.lock");
+    let mut lock_file = File::options()
+        .write(false)
+        .read(true)
+        .open(lock_path)
+        .context("Failed to open lock file")?;
+
+    let mut data: Vec<u8> = Vec::with_capacity(16);
+    lock_file.read_to_end(&mut data)?;
```

Comment:
```suggestion
    let data: Vec<u8> = fs::read(lock_path).context("Failed to read lock file")?;
```

https://doc.rust-lang.org/std/fs/fn.read.html

Reply:
As a sanity check, we could check the `data.len()`?

Reply:
Nice suggestions, thanks!